web/modules/custom/bnf/bnf_client/src/Form/BnfSubscriptionDeleteForm.php

```diff
@@ -0,0 +1,171 @@
+<?php
```

Comment:
You could have saved a lot of code by leaning into the entity system.

Adding the "delete-form" path to the links section of the Subscription entity annotation means you don't have to provide route for it. You could then add "delete" to the "form" handler to point at this class, which in turn could inherit from `ContentEntityDeleteForm` in order to add in our special needs.

Take a look at `Node` and `NodeDeleteForm` in core.

Reply:
I did actually look at it, but I couldn't get it to work.
I think it's fine how it is :)